http/src/test/scala/filodb/http/PrometheusApiRouteSpec.scala

```diff
@@ -114,19 +122,68 @@ class PrometheusApiRouteSpec extends FunSpec with ScalatestRouteTest with AsyncT
     }
   }
 
-    it("should take default spread value if there is no override") {
-      val query = "heap_usage{_ws_=\"demo\",_ns_=\"App-1\"}"
+  it("should take default spread value if there is no override") {
+    val query = "heap_usage{_ws_=\"demo\",_ns_=\"App-1\"}"
 
-      Get(s"/promql/prometheus/api/v1/query_range?query=${query}&" +
-        s"start=1555427432&end=1555447432&step=15&explainOnly=true") ~> prometheusAPIRoute ~> check {
+    Get(s"/promql/prometheus/api/v1/query_range?query=${query}&" +
+      s"start=1555427432&end=1555447432&step=15&explainOnly=true") ~> prometheusAPIRoute ~> check {
+
+      handled shouldBe true
+      status shouldEqual StatusCodes.OK
+      contentType shouldEqual ContentTypes.`application/json`
+      val resp = responseAs[ExplainPlanResponse]
+      resp.status shouldEqual "success"
+      resp.debugInfo.filter(_.startsWith("--E~MultiSchemaPartitionsExec")).length shouldEqual 2
+    }
+  }
+
+  import filodb.query.PromCirceSupport._
+
+  it("should auto convert Histogram data to Prom bucket vectors") {
+    val query = "request-latency{_ws_=\"demo\",_ns_=\"testapp\"}"
+
+    val start = 1555427432
+    val end   = 1555447432
+    Get(s"/promql/prometheus/api/v1/query_range?query=${query}&" +
+      s"start=$start&end=$end&step=15") ~> prometheusAPIRoute ~> check {
 
-        handled shouldBe true
-        status shouldEqual StatusCodes.OK
-        contentType shouldEqual ContentTypes.`application/json`
-        val resp = responseAs[ExplainPlanResponse]
-        resp.status shouldEqual "success"
-        resp.debugInfo.filter(_.startsWith("--E~MultiSchemaPartitionsExec")).length shouldEqual 2
+      handled shouldBe true
+      status shouldEqual StatusCodes.OK
+      contentType shouldEqual ContentTypes.`application/json`
+      val resp = responseAs[SuccessResponse]
+      resp.status shouldEqual "success"
+      resp.data.result should have length (80)    // 8 vectors, one for each bucket * 10 series
+      resp.data.result.foreach { res =>
+        res.value shouldEqual None
+        res.values.get.length should be > (1)
+        res.metric.contains("le") shouldEqual true
+        res.metric("_metric_").endsWith("_bucket") shouldEqual true
+      }
+    }
+  }
+
+  it("should output native Histogram maps if histogramMap=true") {
```

Comment:
Can we add another unit test to ensure that the data points are returned in the correct format:

Today, each sample in json response is represented as 
```
{
"timestamp": 1581099990,
"value": 45.293414
}
```

Some change broke it.

It should have been
```
[
1581099990,
"45.293414"
]
```

Reply:
Yes, this PR should fix this, but it really depends on the encoders implicitly added in Circe, and it is really tricky to read in code and debug.

Reply:
@vishramachandran hey I think it was due to an import removed by IntelliJ by mistake  ;-)
  // DO NOT REMOVE PromCirceSupport import below assuming it is unused - Intellij removes it in auto-imports :( .